src/OT/glyf/glyf.hh

```diff
@@ -349,7 +349,7 @@ struct glyf_accelerator_t
     funcs->push_root_transform (data, &ctx);
 
     funcs->push_clip_glyph (data, gid, &ctx);
-    funcs->color (data, 0xffff, 1., &ctx);
+    funcs->color (data, hb_paint_get_color (&ctx, 0xffff, 1.), &ctx);
     funcs->pop_clip (data, &ctx);
```

Comment:
Hmm, this is not going to work for PDF for me. I output color glyphs to PDF Type 3 glyphs and parts with color index 0xFFFF don’t get any color set and they get the text color when used in PDF text operator i.e. I don’t know the foreground color at the time of writing the layers and the same glyph can be used with different foreground colors.

Reply:
I don’t mind resolving the colors internally, but I need a way to tell when foreground color is to be used without having to set it beforehand.

Reply:
How do you do this for gradients?

Reply:
I've added an is_foreground boolean that you can use. If you pass 0,0,0,1 as the foreground color to hb_font_paint_glyph, you can read off the alpha value in the color.

Reply:
I've added an is_foreground boolean that you can use. If you set the foreground color to 0,0,0,255, then you can read the alpha value off the color.